schema/test/unit/concurrent_revisions/conflicting_creates_test.sql

```diff
@@ -0,0 +1,335 @@
+      'reportingRequirementIndex', 1
+      )::jsonb,
+  null,
+  2
+);
+select cif.add_project_attachment_to_revision(2,1);
+
+-- create the general revision that will be "committing"
+select cif.create_project_revision(1, 'General Revision'); -- id = 3
+select cif.add_contact_to_revision(3, 1, 2);
+select cif.create_form_change(
+  'create',
+  'emission_intensity',
+  'cif',
+  'reporting_requirement',
+  json_build_object(
+      'baselineEmissionIntensity', 5,
+      'targetEmissionIntensity', 6,
+      'postProjectEmissionIntensity', 7,
+      'totalLifetimeEmissionReduction', 8
+      )::jsonb,
+  null,
+  3
+);
+select cif.create_form_change(
+  'create',
+  'funding_parameter_EP',
+  'cif',
+  'funding_parameter',
+  json_build_object(
+      'projectId', 1,
+      'provinceSharePercentage', 2
+      )::jsonb,
+  null,
+  3
+);
+select cif.create_form_change(
+  'create',
+  'project_manager',
+  'cif',
+  'project_manager',
+  json_build_object(
+      'projectManagerLabelId', 1,
+      'cifUserId', 2,
+      'projectId', 1
+      )::jsonb,
+  null,
+  3
+);
+select cif.create_form_change(
+  'create',
+  'project_manager',
+  'cif',
+  'project_manager',
+  json_build_object(
+      'projectManagerLabelId', 2,
+      'cifUserId', 1,
+      'projectId', 1
+      )::jsonb,
+  null,
+  3
+);
+select cif.create_form_change(
+  'create',
+  'reporting_requirement',
+  'cif',
+  'reporting_requirement',
+  json_build_object(
+      'reportType', 'Quarterly',
+      'reportingRequirementIndex', 1,
+      'projectId', 1
+      )::jsonb,
+  null,
+  3
+);
+select cif.create_form_change(
+  'create',
+  'reporting_requirement',
+  'cif',
+  'reporting_requirement',
+  json_build_object(
+      'reportType', 'Quarterly',
+      'reportingRequirementIndex', 2,
+      'projectId', 1
+      )::jsonb,
+  null,
+  3
+);
+select cif.create_form_change(
+  'create',
+  'milestone',
+  'cif',
+  'reporting_requirement',
+  json_build_object(
+      'reportType', 'General Milestone',
+      'reportingRequirementIndex', 1
+      )::jsonb,
+  null,
+  3
+);
+select cif.create_form_change(
+  'create',
+  'milestone',
+  'cif',
+  'reporting_requirement',
+  json_build_object(
+      'reportType', 'General Milestone',
+      'reportingRequirementIndex', 2
+      )::jsonb,
+  null,
+  3
+);
+select cif.add_project_attachment_to_revision(3,2);
+
+select cif.commit_project_revision(3);
+
+select is (
+  (select count(*) from cif.form_change where project_revision_id = 2 and json_schema_name = 'funding_parameter_EP'),
+  1::bigint,
+  'When committing and pending have both created a funding parameter form, only one exists in pending after the first is commit'
+);
+
+select is (
+  (select (new_form_data ->> 'provinceSharePercentage')::int from cif.form_change where project_revision_id = 2 and json_schema_name = 'funding_parameter_EP'),
+  1,
+  'When committing and pending have both created a funding parameter form, pending maintains its form values'
+);
```

Comment:
This is a tricky one. If a value is not set by the amendment in the funding parameter form, but set in the general revision, would a user expect that to be reflected in the amendment?
This is probably a rare edge case where there might be the impression that data was lost. This can probably be resolved with some training.

Reply:
It's a very good question. I leaned towards "the amendment is right" when I wasn't sure but I'd love to know where Business Area sit on the specific cases.

Reply:
Probably careful examination of the changes before submitting an amendment or a revision would do the trick